app/views/unattended/provisioning_templates/host_init_config/host_init_config_default.erb

```diff
@@ -10,10 +10,27 @@ oses:
 -%>
 #!/bin/bash
 
-set -e
+<% built_https = foreman_url('built').start_with?('https') -%>
 
 # This script enrolls a host with Foreman.
 # Host: <%= @host.name %>
+echo "# Running [<%= @host.name %>] host initial configuration"
+
+trap 'exit_and_cancel_build' ERR
```

Comment:
I feel like defining the trap if the function hasn't been defined is odd. What if writing the SSL cert fails for some reason? Should the trap be right after the `exit_and_cancel_build` definition?

Also makes me wonder if we should use `set -e` until the trap has been set, then use `set +e` to switch over.